envoy/ssl/context.h

```diff
@@ -21,9 +21,10 @@ class Context {
   virtual ~Context() = default;
 
   /**
-   * @return the number of days in this context until the next certificate will expire
+   * @return the number of days in this context until the next certificate will expire, the value is
+   * set when not expired.
    */
-  virtual size_t daysUntilFirstCertExpires() const PURE;
+  virtual absl::optional<int32_t> daysUntilFirstCertExpires() const PURE;
```

Comment:
Why switch back to signed integers if we never return a negative number?

Reply:
SIZE_T is enough, I will change it back.

Reply:
`size_t` is a 64-bit integer on many architectures. JSON numbers are actually 64-bit floats, so large values of size_t cannot be exactly represented, which causes the test failure that you fixed by switching to `int32_t`. Instead, I think you should use `uint32_t` and `std::numeric_limits<uint32_t>::max()` where you need the maximum value.

Reply:
Oh, I get it, thanks for your patience in explaining.